src/main/java/com/redislabs/redisai/Script.java

```diff
@@ -46,56 +50,55 @@ public Script(Device device, String source) {
    *
    * @param device the device that will execute the model. can be of CPU or GPU
    * @param filePath file path to load the script from
+   * @throws java.io.IOException
    */
   public Script(Device device, Path filePath) throws IOException {
-    this(device);
-    this.source =
-        Files.readAllLines(filePath, StandardCharsets.UTF_8).stream()
-                .collect(Collectors.joining("\n"))
-            + "\n";
+    this(device, fileContent(filePath));
+  }
+
+  private static String fileContent(Path filePath) throws IOException {
+    return Files.readAllLines(filePath, StandardCharsets.UTF_8).stream()
+            .collect(Collectors.joining("\n"))
+        + "\n";
   }
 
   public static Script createScriptFromRespReply(List<?> reply) {
-    Script script = null;
-    String source = null;
     Device device = null;
     String tag = null;
+    String source = null;
+    List<String> entryPoints = null;
     for (int i = 0; i < reply.size(); i += 2) {
-      String arrayKey = SafeEncoder.encode((byte[]) reply.get(i));
-      switch (arrayKey) {
+      String mapKey = SafeEncoder.encode((byte[]) reply.get(i));
+      Object mapVal = reply.get(i + 1);
+      switch (mapKey) {
         case "source":
-          source = SafeEncoder.encode((byte[]) reply.get(i + 1));
+          source = BuilderFactory.STRING.build(mapVal);
           break;
         case "device":
-          String deviceString = SafeEncoder.encode((byte[]) reply.get(i + 1));
-          device = Device.valueOf(deviceString);
-          if (device == null) {
-            throw new JRedisAIRunTimeException("Unrecognized device: " + deviceString);
-          }
+          device = Device.valueOf(BuilderFactory.STRING.build(mapVal));
           break;
         case "tag":
-          tag = SafeEncoder.encode((byte[]) reply.get(i + 1));
+          tag = BuilderFactory.STRING.build(mapVal);
+          break;
+        case "Entry Points":
+          entryPoints = BuilderFactory.STRING_LIST.build(mapVal);
           break;
         default:
           break;
       }
     }
     if (device != null && source != null) {
-      script = new Script(device, source);
-      if (tag != null) {
-        script.setTag(tag);
-      }
-    } else {
-      throw new JRedisAIRunTimeException(
-          "AI.SCRIPTGET reply did not contained all elements to build the script");
+      return new Script(device, source).setTag(tag).setEntryPoints(entryPoints);
     }
-    return script;
+    throw new JRedisAIRunTimeException(
+        "AI.SCRIPTGET reply did not contained all elements to build the script");
   }
 
   public Device getDevice() {
     return device;
   }
 
+  @Deprecated
```

Comment:
why?

Reply:
@DvirDukhan Device is a mandatory param for a Script. I'm thinking about setting it from only constructor.